buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/TestClustersPlugin.java

```diff
@@ -123,7 +120,7 @@ private void createListClustersTask(Project project, NamedDomainObjectContainer<
     private void createUseClusterTaskExtension(Project project, NamedDomainObjectContainer<ElasticsearchCluster> container) {
         // register an extension for all current and future tasks, so that any task can declare that it wants to use a
         // specific cluster.
-        project.getTasks().all((Task task) ->
+        project.getTasks().withType(TestClustersTask.class, (Task task) ->
```

Comment:
We should try and align these two methods somehow at some point. That is, calling `testCluster` or `useCluster` would be all that is needed to register all the necessary wiring here. We could something like have the `testCluster` method on `TestClustersTask` reach out and grab the test clusters plugin and perform the registration.

We don't have to do this as part of this PR though.